oraclesrv/utils.py

```diff
@@ -142,3 +150,104 @@ def get_solr_data_match_thesis(author, year, doctype):
         status_code = e.response.status_code
 
     return result, query, status_code
+
+def add_a_record(docmatch):
+    """
+
+    :param docmatch:
+    :return:
+    """
+    # self.logger.debug("Added a `Reference` record successfully.")
+    # return reference.bibcode, reference.source_filename
+
+    try:
+        with current_app.session_scope() as session:
+            found = session.query(exists().where(and_(DocMatch.source_bibcode == docmatch.source_bibcode,
```

Comment:
i can imagine situations when the score changes (even just because of different machine/hardware and floating point rounding errors) - without deltas, this may result in nothing found (when a close match is there). Not sure if the `confidence` should even be part of the query

and i have other observations about 'source_bibcode:matched_bibcode` which i'll write into general text

Reply:
to the question of confidence being part of query here, I think it should. Here is to add a new record, if it is new, to the database, so we need to find out if the exact record exists or not. There can be multiple source/matched with different confidence, as I have explained in the DL document, it is possible that a true match is not in ADS yet, so the best match has a high enough confidence to be added to db. Then when the true match is in ADS and confidence is very high, that record needs to be added to. So the uniqueness is source/matched/confidence.